src/main/java/org/openrewrite/java/testing/cucumber/RegexToCucumberExpression.java

```diff
@@ -0,0 +1,117 @@
+
+    private static final String IO_CUCUMBER_JAVA_STEP_DEFINITION = "io.cucumber.java.*.*";
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesType<>(IO_CUCUMBER_JAVA_STEP_DEFINITION);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace Cucumber-Java step definition regexes with Cucumber expressions.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Strip regex prefix and suffix from step annotation expressions arguments where possible.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new CucumberStepDefinitionBodyVisitor();
+    }
+
+    static final class CucumberStepDefinitionBodyVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration m, ExecutionContext p) {
+            J.MethodDeclaration methodDeclaration = super.visitMethodDeclaration(m, p);
+            return methodDeclaration.withLeadingAnnotations(ListUtils.map(methodDeclaration.getLeadingAnnotations(),
+                    ann -> replaceRegexWithCucumberExpression(methodDeclaration, ann, p)));
+        }
+
+        private static J.Annotation replaceRegexWithCucumberExpression(
+                // For when we want to match regexes with method arguments for replacement cucumber expressions
+                // https://github.com/cucumber/cucumber-expressions#parameter-types
+                J.MethodDeclaration methodDeclaration,
+                J.Annotation annotation,
+                ExecutionContext p) {
+            List<Expression> arguments = annotation.getArguments();
+            Optional<String> possibleExpression = Stream.of(arguments)
+                    .filter(Objects::nonNull)
+                    .filter(list -> list.size() == 1)
+                    .flatMap(Collection::stream)
+                    .filter(J.Literal.class::isInstance)
+                    .map(e -> (J.Literal) e)
+                    .map(l -> (String) l.getValue())
+                    // https://github.com/cucumber/cucumber-expressions/blob/main/java/heuristics.adoc
+                    .filter(s -> (s.startsWith("^") || s.endsWith("$"))
+                            || (s.startsWith("/") && s.endsWith("/")))
+                    .findFirst();
+            if (!possibleExpression.isPresent()) {
+                return annotation;
+            }
+
+            // Strip leading/trailing regex anchors
+            String replacement = possibleExpression.get();
+            if (replacement.startsWith("^") || replacement.startsWith("/")) {
+                replacement = replacement.substring(1);
+            }
+            if (replacement.endsWith("$") || replacement.endsWith("/")) {
```

Comment:
IIRC slashes are only a strong indicator of a Regex when they're on both ends.

Reply:
That's already handled by the filter on line 89 right? `(s.startsWith("/") && s.endsWith("/"))`

Reply:
Not quite.

Consider `^hello world/`, this should map to `hello world/`, not `hello world`.

Reply:
Scherp; https://github.com/openrewrite/rewrite-testing-frameworks/pull/265/commits/fd60346bc3274eb76850c67f4dd708b7eaf821e1